sky/clouds/aws.py

```diff
@@ -477,51 +480,91 @@ def _get_image_id(
-                # Fallback to default image size if no credentials are available.
-                # The credentials issue will be caught when actually provisioning
-                # the instance and appropriate errors will be raised there.
-                return DEFAULT_AMI_GB
+                    f'Failed to get {log_context} for {image_id} in region '
+                    f'{region}: {e}. Using default value.')
+                return None
             except aws.botocore_exceptions().ClientError as e:
-                logger.debug(
-                    f'Failed to get image size for {image_id} in region {region}: {e}'
-                )
+                # This shared log message replaces two attribute-specific
+                # messages (image size/root device) for simplicity.
+                logger.debug(f'Failed to get {log_context} for image '
+                             f'{image_id!r} in region {region}: {e}')
                 if iteration == max_retries:
                     with ux_utils.print_exception_no_traceback():
+                        if env_options.Options.SHOW_DEBUG_INFO.get():
+                            image_not_found_message += f'\n{debug_message}'
+                            # Note: the ClientError's exception message should
+                            # include most useful info:
+                            # https://github.com/boto/botocore/blob/260a8b91cedae895165984d2102bcbc487de3027/botocore/exceptions.py#L518-L532
+                            additional_info = f'  ClientError: {e}'
+                            logger.debug(additional_info)
+                            image_not_found_message += '\n' + additional_info
                         raise ValueError(image_not_found_message) from None
             # linear backoff starting from 0.5 seconds
             time.sleep(iteration * 0.5)
+        # Should never reach here, but keep type checker happy.
+        raise RuntimeError('Unreachable')
+
+    @classmethod
+    def get_image_size(cls, image_id: str, region: Optional[str]) -> float:
+        if image_id.startswith('skypilot:'):
+            return DEFAULT_AMI_GB
+        assert region is not None, (image_id, region)
+        # first try the cache
+        workspace_profile = aws.get_workspace_profile()
+        kv_cache_key = f'aws:ami:size:{workspace_profile}:{region}:{image_id}'
+        image_size = kv_cache.get_cache_entry(kv_cache_key)
+        if image_size is not None:
+            return float(image_size)
+        # if not found in cache, query the cloud
+        image = cls._describe_image_with_retry(
+            image_id,
+            region,
+            log_context='image size',
+        )
+        if image is None:
+            # Fallback to default image size if no credentials are available.
+            # The credentials issue will be caught when actually provisioning
+            # the instance and appropriate errors will be raised there.
+            return DEFAULT_AMI_GB
```

Comment:
Add a debug log for the missing image from describe?

Reply:
It's already in `_describe_image_with_retry`:
https://github.com/cg505/skypilot/blob/23dce140ed0090ae9fcdcb3aee853bd999f911cd/sky/clouds/aws.py#L516-L523